### PR TITLE
chore(refactor): finish passing npm context

### PR DESCRIPTION
**Pull request by wraithgar**
No more requiring npm as a singleton.  This will now allow us to move
forward with the other refactoring we need to always use the npm object
itself in tests, not a mocked one.

This constitutes an extreme paring down of the logic in the error-message code, removing the untested cache-file.js and moving what we actually needed of its logic inline into error-message.js.

## References
Closes https://github.com/npm/statusboard/issues/279

### CODE REVIEW

**Comment by wraithgar** [comment]
Still needs tests for the code in cache-file that had never been tested, and was moved into error-handler itself.

File: tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs [hunk 1]
```diff
@@ -208,31 +190,7 @@ to accept an incorrect (and potentially broken) dependency resolution.
 See \${REPORT} for a full report.
 `
 
-exports[`test/lib/utils/explain-eresolve.js TAP cycleNested > report with color, depth only 2 1`] = `
```
Comment (wraithgar) [inline tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs:211]: Need to fix this still

File: tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs [hunk 2]
```diff
@@ -366,29 +324,7 @@ to accept an incorrect (and potentially broken) dependency resolution.
 See \${REPORT} for a full report.
 `
 
-exports[`test/lib/utils/explain-eresolve.js TAP gatsby > report with color, depth only 2 1`] = `
```
Comment (wraithgar) [inline tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs:369]: Need to fix this

File: tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs [hunk 3]
```diff
@@ -480,23 +415,7 @@ to accept an incorrect (and potentially broken) dependency resolution.
 See \${REPORT} for a full report.
 `
 
-exports[`test/lib/utils/explain-eresolve.js TAP no current node, but has current edge > report with color, depth only 2 1`] = `
```
Comment (wraithgar) [inline tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs:483]: Fix

File: tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs [hunk 4]
```diff
@@ -570,22 +489,7 @@ to accept an incorrect (and potentially broken) dependency resolution.
 See \${REPORT} for a full report.
 `
 
-exports[`test/lib/utils/explain-eresolve.js TAP no current node, no current edge, idk > report with color, depth only 2 1`] = `
```
Comment (wraithgar) [inline tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs:573]: Fix

File: tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs [hunk 5]
```diff
@@ -677,26 +581,7 @@ to accept an incorrect (and potentially broken) dependency resolution.
 See \${REPORT} for a full report.
 `
 
-exports[`test/lib/utils/explain-eresolve.js TAP withShrinkwrap > report with color, depth only 2 1`] = `
```
Comment (wraithgar) [inline tap-snapshots/test/lib/utils/explain-eresolve.js.test.cjs:680]: Fix

**Comment by wraithgar** [comment]
The explain-eresolve snapshot diff looks like it's replacing tests w/ empty output, but if you inspect the file itself they are correct.  There were tests removed due to the fact that report doesn't accept a depth parameter (since nothing was passing one in the codebase)

**Review by ruyadorno — APPROVED** [review]

**Pull request merged**